src/api/app/controllers/webui/users/tokens/groups_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Webui::Users::Tokens::GroupsController < Webui::WebuiController
+  def create
+    token = Token::Workflow.find(params[:token_id])
+    group = Group.find_by(title: params[:groupid])
+
+    if token.groups_shared_among.include?(group)
+      redirect_back(fallback_location: root_path, notice: "Group #{group.title} is already associated to the token")
+    elsif token.groups_shared_among << group
+      redirect_back(fallback_location: root_path, success: "Group #{group.title} has been added to the token successfully")
+    end
```

Comment:
Not sure why you would error on this? Just don't do anything, what the user requested is already done...

Reply:
This is not an error but a 'notice', just an informative blue flash message.

Reply:
Do you still prefer to do nothing?

Reply:
Indeed. Less, more understandable code.